cmd/mattermost/commands/server.go

```diff
@@ -45,15 +45,12 @@ func serverCmdF(command *cobra.Command, args []string) error {
 
 func runServer(configFileLocation string, disableConfigWatch bool, usedPlatform bool, interruptChan chan os.Signal) error {
 	options := []app.Option{
-		app.ConfigFile(configFileLocation),
+		app.ConfigFile(configFileLocation, !disableConfigWatch),
```

Comment:
The watcher will be built into the new file store (and otherwise remain completely hidden), so this simplifies setup vs. having a separate `Option`.